muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/dialogs/student/tabs/guidance-relation.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from "react";
+import { i18nType } from "~/reducers/base/i18n";
+import { StateType } from "~/reducers";
+import { connect, Dispatch } from "react-redux";
+import { bindActionCreators } from "redux";
+import { AnyActionType } from "~/actions";
+import ApplicationSubPanel, {
+  ApplicationSubPanelViewHeader,
+  ApplicationSubPanelItem,
+} from "~/components/general/application-sub-panel";
+import { GuiderStudentUserProfileType } from "~/reducers/main-function/guider";
+import ContactEvent from "./contact-events/contact-event";
+import NewContactEvent from "./contact-events/editors/new-event";
+import { ButtonPill } from "~/components/general/button";
+import PagerV2 from "~/components/general/pagerV2";
+import {
+  loadStudentContactLogs,
+  LoadContactLogsTriggerType,
+} from "~/actions/main-function/guider";
+
+/**
+ * GuidanceRelationProps
+ */
+interface GuidanceRelationProps {
+  i18n: i18nType;
+  currentStudent: GuiderStudentUserProfileType;
+  contactLogsPerPage: number;
+  loadStudentContactLogs: LoadContactLogsTriggerType;
+}
+/**
+ * Context for the contactLogsPerPage
+ */
+export const ContactLogsContext = React.createContext(10);
```

Comment:
This could need more specific explanation what is it for? Why and what is it for? Yeah sure by reading code more you find out that its for how many contact logs one page have.